qiskit_dynamics/models/generator_models.py

```diff
@@ -311,137 +276,94 @@ def frame(self) -> Frame:
 
     @frame.setter
     def frame(self, frame: Union[Operator, Array, Frame]):
-        """Set the frame; either an already instantiated :class:`Frame` object
-        a valid argument for the constructor of :class:`Frame`, or `None`.
-        """
+        if self._frame is not None and self._frame.frame_diag is not None:
+            self._operator_collection.drift = self._operator_collection.drift + Array(
+                np.diag(self._frame.frame_diag)
+            )
+            self._operator_collection.apply_function_to_operators(
+                self.frame.operator_out_of_frame_basis
+            )
 
         self._frame = Frame(frame)
 
-        self._reset_internal_ops()
-
-    @property
-    def cutoff_freq(self) -> float:
-        """Return the cutoff frequency."""
-        return self._cutoff_freq
-
-    @cutoff_freq.setter
-    def cutoff_freq(self, cutoff_freq: float):
-        """Set the cutoff frequency."""
-        if cutoff_freq != self._cutoff_freq:
-            self._cutoff_freq = cutoff_freq
-            self._reset_internal_ops()
-
-    def evaluate(self, time: float, in_frame_basis: bool = False) -> Array:
-        """Evaluate the model in array format.
+        if self._frame.frame_diag is not None:
+            self._operator_collection.apply_function_to_operators(
+                self.frame.operator_into_frame_basis
+            )
+            self._operator_collection.drift = self._operator_collection.drift - Array(
+                np.diag(self._frame.frame_diag)
+            )
 
+    def evaluate_without_state(self, time: float, in_frame_basis: Optional[bool] = False) -> Array:
+        """Evaluate the model in array format as a matrix, independent of state.
         Args:
             time: Time to evaluate the model
             in_frame_basis: Whether to evaluate in the basis in which the frame
                             operator is diagonal
-
         Returns:
-            Array: the evaluated model
-
+            Array: the evaluated model as a (n,n) matrix
         Raises:
-            QiskitError: If model cannot be evaluated.
-        """
+            QiskitError: If model cannot be evaluated."""
 
         if self._signals is None:
             raise QiskitError("""GeneratorModel cannot be evaluated without signals.""")
 
-        sig_vals = self._signals.complex_value(time)
+        sig_vals = np.real(self._signals.complex_value(time))
```

Comment:
For this you can directly call `self._signals(time)` (as it will evaluate the real part).